File: app/src/main/java/com/example/bangabandhuplay/ui/movies/MoviesContentAdapter.java
package com.example.bangabandhuplay.ui.movies;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.navigation.NavController;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.bangabandhuplay.data.model.category.root.single.OttContent;
import com.example.bangabandhuplay.data.model.category.root.single.SubCategory;
import com.example.bangabandhuplay.databinding.SectionRecyclerViewLayoutBinding;

import java.util.List;

public class MoviesContentAdapter extends RecyclerView.Adapter<MoviesContentAdapter.ParentViewHolder> {


    private List<SubCategory> subCategories;

    private Context context;


    public MoviesContentAdapter(List<SubCategory> subCategories,  Context context) {
        this.subCategories = subCategories;
        this.context = context;
    }

    @NonNull
    @Override
    public ParentViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ParentViewHolder(SectionRecyclerViewLayoutBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ParentViewHolder holder, int position) {

        SubCategory current = subCategories.get(position);

        List<OttContent> ottContents = current.getOttContents();

        /*holder.binding.itemRv.setLayoutManager(new LinearLayoutManager(context, RecyclerView.HORIZONTAL, false));
        holder.binding.itemRv.setAdapter(new MoviesChildContentAdapter(ottContents));*/

        String title = current.getTitle();

        if (title != null){
            holder.binding.titleTv.setText(title);
        }

        holder.binding.moreTv.setOnClickListener(view -> {
        });



    }

    @Override
    public int getItemCount() {
        return subCategories != null ? subCategories.size() : 0;
    }

    class ParentViewHolder extends RecyclerView.ViewHolder {
        SectionRecyclerViewLayoutBinding binding;

        ParentViewHolder(SectionRecyclerViewLayoutBinding itemView) {
            super(itemView.getRoot());
            binding = itemView;
        }
    }

    public interface CallBack{
        void addWishListItem(String uuid);
    }
}

File: app/src/main/java/live/durbar/bangabandhuapp/ui/movies/MoviesContentAdapter.kt
package live.durbar.bangabandhuapp.ui.movies

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import live.durbar.bangabandhuapp.data.model.category.root.single.SubCategory
import live.durbar.bangabandhuapp.databinding.SectionRecyclerViewLayoutBinding
import live.durbar.bangabandhuapp.ui.more.MoreActivity
import live.durbar.bangabandhuapp.utils.Constants

class MoviesContentAdapter(private val subCategories: List<SubCategory>, private val context: Context, private var isTunes: Boolean) :
    RecyclerView.Adapter<MoviesContentAdapter.ParentViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ParentViewHolder {
        return ParentViewHolder(
            SectionRecyclerViewLayoutBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        )
    }

    override fun onBindViewHolder(holder: ParentViewHolder, position: Int) {
        val current = subCategories[position]
        val title = current.title
        val slug = current.slug
        val ottContents = current.ottContents

        holder.binding.itemRv.layoutManager =
            LinearLayoutManager(context, RecyclerView.HORIZONTAL, false)
        holder.binding.itemRv.adapter = ottContents?.let { MoviesChildContentAdapter(it, context, title?:"") }

        title?.let { holder.binding.titleTv.text = it }

        holder.binding.moreTv.setOnClickListener {
            if (slug != null && title != null) {
                context.startActivity(
                    Intent(context, MoreActivity::class.java)
                        .putExtra(Constants.CONTENT_SLUG, slug)
                        .putExtra(Constants.CONTENT_SECTION_TITLE, title)
                        .putExtra(Constants.CONTENT_IS_HOME, false)
                )
                Constants.setEditor(context, Constants.CONTENT_SLUG, slug)
                Constants.setEditor(context, Constants.CONTENT_SECTION_TITLE, title)
                Constants.setEditor(context, Constants.CONTENT_IS_HOME, false)
            }
        }
        if (isTunes){
            Constants.IS_TUNES = true
        }else{
            Constants.IS_TUNES = false

        }
    }

    override fun getItemCount(): Int {
        return subCategories.size
    }

    inner class ParentViewHolder(val binding: SectionRecyclerViewLayoutBinding) :
        RecyclerView.ViewHolder(binding.root)
}
